pkg/testcase/privateregistry.go

```diff
@@ -0,0 +1,87 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/pkg/helper"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+
+func TestBuildPrivateCluster(cluster *shared.Cluster) {
+	Expect(cluster.Status).To(Equal("cluster created"))
+	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
+
+	if cluster.GeneralConfig.BastionIP != "" {
+		shared.LogLevel("info", "Bastion Node IP: " + cluster.GeneralConfig.BastionIP)
+		shared.LogLevel("info", "Bastion Node DNS: " + cluster.GeneralConfig.BastionDNS)
+	}
+	shared.LogLevel("info", "Server Node IPs: %v", cluster.ServerIPs)
+
+	checkAndPrintAgentNodeIPs(cluster.NumAgents, cluster.AgentIPs, false)
+
+	if cluster.Config.Product == "rke2" {
+		checkAndPrintAgentNodeIPs(cluster.NumWinAgents, cluster.WinAgentIPs, true)
+	}
+}
+
+func TestPrivateRegistry(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	serverFlags := os.Getenv("server_flags")
+	agentFlags := os.Getenv("worker_flags")
+	helper.SetupBastion(cluster, flags)
+	helper.CopyAssetsOnNodes(cluster)
+	var token string
+	for idx, serverIP := range cluster.ServerIPs {
+		if idx == 0 {
+			shared.LogLevel("info", "Installing %v on server node-1...", cluster.Config.Product)
+			cmd := fmt.Sprintf(
+				"sudo chmod +x install_product.sh; "+
+					`sudo ./install_product.sh "%v" "" "" "server" "%v" "%v"`,
+				cluster.Config.Product, serverIP, serverFlags)
+			_, err := helper.CmdForPrivateNode(cluster, cmd, serverIP)
+			Expect(err).To(BeNil())
+
+			cmd = fmt.Sprintf("sudo cat /var/lib/rancher/%v/server/token", cluster.Config.Product)
+			token, err = helper.CmdForPrivateNode(cluster, cmd, serverIP)
+			Expect(err).To(BeNil())
+			Expect(token).NotTo(BeEmpty())
+			shared.LogLevel("info", "token: " + token)
```

Comment:
why logging the token ?

Reply:
for debugging purpose in case the installation on other node fails, will use debug level

Reply:
Updated